### PR TITLE
Support Succeeded Condition

### PR DESCRIPTION
**Pull request by mshitrit**
In order to fix ticket [ECOPROJECT-1246](https://issues.redhat.com//browse/ECOPROJECT-1246) ,and in context of NHC PR: https://github.com/medik8s/node-healthcheck-operator/pull/208

This PR add the support to Succeeded Condition in SNR

### CODE REVIEW

File: controllers/selfnoderemediation_controller.go [hunk 1]
```diff
@@ -212,27 +212,40 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
...
+		succeededConditionStatus = metav1.ConditionFalse
 	default:
-		err := fmt.Errorf("unkown processingChangeReason:%s", reason)
+		err := fmt.Errorf("unkown processingChangeReason:%s", processingTypeReason)
 		r.Log.Error(err, "couldn't update snr processing condition")
 		return err
 	}
 
-	if meta.IsStatusConditionPresentAndEqual(snr.Status.Conditions, v1alpha1.SnrConditionProcessing, conditionStatus) {
+	if meta.IsStatusConditionPresentAndEqual(snr.Status.Conditions, v1alpha1.ProcessingTypeCondition, processingConditionStatus) {
```
Comment (slintes) [inline controllers/selfnoderemediation_controller.go:233]: what about the Succeeded condition?

File: api/v1alpha1/selfnoderemediation_types.go [hunk 2]
```diff
@@ -23,8 +23,10 @@ import (
 const (
 	ResourceDeletionRemediationStrategy  = RemediationStrategyType("ResourceDeletion")
 	OutOfServiceTaintRemediationStrategy = RemediationStrategyType("OutOfServiceTaint")
-	// SnrConditionProcessing is the condition type used to signal NHC the remediation status
-	SnrConditionProcessing = "Processing"
+	// ProcessingTypeCondition is the condition type used to signal NHC the remediation status
+	ProcessingTypeCondition = "Processing"
```
Comment (slintes) [inline api/v1alpha1/selfnoderemediation_types.go:27]: somehow this reads a bit strange... would either ProcessingConditionType or ConditionTypeProcessing be a better fit?

File: controllers/selfnoderemediation_controller.go [hunk 3]
```diff
@@ -212,27 +212,40 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
 	return result, r.updateSnrStatusLastError(snr, err)
 }
 
-func (r *SelfNodeRemediationReconciler) updateSnrProcessingCondition(reason processingChangeReason, snr *v1alpha1.SelfNodeRemediation) error {
-	var conditionStatus metav1.ConditionStatus
-	switch reason {
+func (r *SelfNodeRemediationReconciler) updateSnrTypeConditions(processingTypeReason processingChangeReason, snr *v1alpha1.SelfNodeRemediation) error {
```
Comment (slintes) [inline controllers/selfnoderemediation_controller.go:215]: this also reads strange.... maybe just updateConditions()? (IMHO its clear that it is about SNR CRs, and conditions have a type)

**Review by slintes — CHANGES_REQUESTED** [review]
two nits about naming, but also one issue

File: controllers/selfnoderemediation_controller.go (hunk 1 above)
Comment (mshitrit) [reply controllers/selfnoderemediation_controller.go:233]: Maybe I'm missing something but I did think about it. 
Originally the idea was to skip update if Processing condition was not change, logically IIUC there is no such use case where Success condition would change and Processing condition will not so current condition still good enough.

Comment (slintes) [reply controllers/selfnoderemediation_controller.go:233]: remediationFinished and remediationTerminatedByNHC hace the same Processing but different Succeeeded status?

Comment (mshitrit) [reply controllers/selfnoderemediation_controller.go:233]: They do, but this is not my point.
The point that I'm arguing is that Succedded condition will change only when Processing status also changes.

Comment (slintes) [reply controllers/selfnoderemediation_controller.go:233]: >  Succeeded will change only when Processing also changes

when we first have remediationFinished and then remediationTerminatedByNHC or the other way around, Processing will stay at False, but Succeeded changes. Even when we know that this transition is unlikely or even impossible to happen today, that might change. Or we can get new reasons, resulting in other status combinations. IMHO this is is a maintenance burden and/or a potential future bug.

Comment (mshitrit) [reply controllers/selfnoderemediation_controller.go:233]: IMO both are fine, on one hand I don't see an issue with the current code but on the other hand the code change is very simple and risk free so I'll be happy to oblige :smile:

**Review by slintes — APPROVED** [review]

**Comment by mshitrit** [comment]
Already passed
/override ci/prow/4.13-openshift-e2e

**Pull request merged**